components/Valuation/DataComparison/DataComparisonCard.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { IAPIData, IPredictions } from "../../../lib/types";
+
+interface Props {
+  apiData: IAPIData | undefined;
+  predictions: IPredictions | undefined;
+}
+
+function getMetaverseTokenFromOpenseaLink(opensea_link: string | undefined) {
+  let metaverse = "";
+  let tokenID = "";
+  if (opensea_link) {
+    const split = opensea_link.split("/");
+    metaverse = split[4];
+    tokenID = split[5];
+  }
+  return { metaverse, tokenID };
+}
+
+/**
+ * 
+ * @param param0 
+ * @returns 
+ */
+const DataComparisonCard = ({ apiData, predictions }: Props) => {
+  const [offers, setOffers] = useState<number>();
+  const [openseaLink, setOpenSeaLink] = useState<string | undefined>(apiData?.external_link);
+  
+  const usdPredictionPrice = predictions?.usdPrediction;
+  const handleData = async () => {
+    let { metaverse, tokenID } = getMetaverseTokenFromOpenseaLink(
+      apiData?.opensea_link
+    );
+    const res = await fetch("/api/getLandOffers", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ metaverse: metaverse, tokenId: tokenID }),
+    });
+    const data = await res.json();
+    return data;
+  };
+
+  let usdPrice;
+  let offerPrice;
```

Comment:
Lets try to keep things as scoped and less global as possible (unless needed) This makes the code easier to understand for other devs, especially once components go very big. Unless you are using this two variables outside of the `useEffect` just declare them in the use effect, and you also could just declare them on lines 53 and 55 as well and save this two lines of code here.